operator/v1/types_console.go

```diff
@@ -106,6 +106,38 @@ type ConsoleCustomization struct {
 	// SVG format preferred
 	// +optional
 	CustomLogoFile configv1.ConfigMapFileReference `json:"customLogoFile,omitempty"`
+	// developerCatalog allows to configure the shown developer catalog categories.
+	// +optional
+	DeveloperCatalog DeveloperConsoleCatalogCustomization `json:"developerCatalog,omitempty"`
+}
+
+// DeveloperConsoleCatalogCustomization allow the cluster admin to configure developer catalog.
+type DeveloperConsoleCatalogCustomization struct {
+	// categories which are shown the in the devconsole.
+	// +optional
+	Categories []DeveloperConsoleCatalogCategory `json:"categories,omitempty"`
+}
+
+// DeveloperConsoleCatalogCategoryMeta defines a category incl. label and filter tags
+// for the developer console catalog.
+type DeveloperConsoleCatalogCategoryMeta struct {
+	// ID is an identifier used in the URL to enable deep linking in console.
```

Comment:
We should put some restrictions on the allowed characters.

Reply:
```
	// +kubebuilder:validation:Required
        // +kubebuilder:validation:MinLength=1
	// +required
```

Reply:
All type properties have now

```
	// +kubebuilder:validation:Optional
	// +optional
```

or

```
	// +kubebuilder:validation:Required
	// +required
```

@jhadvig Is MinLength really required if a field is required?

Reply:
If we intend this to be used in a URL, we might want a more restrictive regex like used for k8s resource names.